hkmc2/shared/src/test/mlscript-compile/LazyArray.mls

```diff
@@ -4,18 +4,52 @@ import "./Iter.mls"
 open Iter
 open LazyArray
 
+// Helper Methods
 fun normIdx(i, len) =
   if i < 0 then len + i
   else i
 
+fun concatH(a, l, ...args) =
+  let idx = 0
+  let init_len = a.length
+  set a.length = init_len + args.length * 2
+  while (idx < args.length) do
+    let x = args.[idx]
+
+    set a.[init_len + idx * 2] = SpliceMarker
+    set a.[init_len + idx * 2 + 1] = x
+    set l += x.length
+    set idx += 1
+  Splice of a, l
+
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice
+fun normIdxSlice(i, len) =
+  if i < 0 and
+    i >= -len
+    then len + i
+    else 0
+  else i
+
+fun sliceH(beg, fin, arr) =
```

Comment:
What's this mysterious `H` suffix for?

~~This is not Haskell.~~

Reply:
They're just helpers - I'll rename them `xxxHelper` :sweat_smile: